Scripts/build-phases/swiftlint.sh

```diff
@@ -34,22 +37,22 @@ fi
 # The `|| true` at the end is to stop `grep` from returning a non-zero exit if there
 # are no matches. Xcode's build will fail if we don't do this.
 #
-MODIFIED_FILES=`git diff --name-only --diff-filter=d HEAD | grep -G "\.swift$" || true`
-echo $MODIFIED_FILES | xargs $SWIFTLINT --config $CONFIG_FILE --quiet
+MODIFIED_FILES=$(git diff --name-only --diff-filter=d HEAD | grep -G "\.swift$" || true)
+echo "$MODIFIED_FILES" | xargs $SWIFTLINT_BIN "$SWIFTLINT_ARGS"
 MODIFIED_FILES_LINT_RESULT=$?
 
 # Run SwiftLint on the added files
-ADDED_FILES=`git ls-files --others --exclude-standard | grep -G "\.swift$" || true`
-echo $ADDED_FILES | xargs $SWIFTLINT --config $CONFIG_FILE --quiet
+ADDED_FILES=$(git ls-files --others --exclude-standard | grep -G "\.swift$" || true)
+echo "$ADDED_FILES" | xargs $SWIFTLINT_BIN "$SWIFTLINT_ARGS"
 ADDED_FILES_LINT_RESULT=$?
 
 # Restore the previous directory
-popd > /dev/null
+popd > /dev/null || exit 1
 
-# Exit with non-zero if SwiftLint found a serious violation in the linted files. 
+# Exit with non-zero if SwiftLint found a serious violation in the linted files.
 #
 # This stops Xcode from complaining about "...did not return a nonzero exit code...".
 #
-if [ $MODIFIED_FILES_LINT_RESULT -ne 0 ] || [ $ADDED_FILES_LINT_RESULT -ne 0 ] ; then 
+if [ $MODIFIED_FILES_LINT_RESULT -ne 0 ] || [ $ADDED_FILES_LINT_RESULT -ne 0 ] ; then
```

Comment:
Here's a screenshot showing the changes to the script do not affect SwiftLint's behavior:

![image](https://github.com/woocommerce/woocommerce-ios/assets/1218433/4e425e67-7e9f-4c03-8613-d5cacc3ceded)

I briefly considered modifying this to run on all files, but I found it was quite slow. SwiftLint supports caching which we might look into in a follow up PR.

With diff only, it run in 4-5 seconds.

![image](https://github.com/woocommerce/woocommerce-ios/assets/1218433/1104cfc2-2105-4d28-9b6f-eb6a3baefaec)